cmd/root.go

```diff
@@ -22,39 +22,48 @@ THE SOFTWARE.
 package cmd
 
 import (
-	"fmt"
 	"os"
 	"starq/internal/starq"
 
 	"github.com/spf13/cobra"
 )
 
-// rootCmd represents the base command when called without any subcommands
-var rootCmd = &cobra.Command{
-	Use:   "starq [-r <jq-rule> ...] [-c <config-file>.yaml] [-a <jq-rule> ...]",
-	Short: "starq transforms a document from stdin using jq-style rules.",
-	Long:  `starq (pronounced "star-q") transforms a JSON or YAML document from stdin using jq-style rules and outputs to stdout.`,
-	Run: func(cmd *cobra.Command, args []string) {
-		err := cmd.ParseFlags(args)
-		exitIf(err)
+// NewRootCmd creates a new instance of the base command without any subcommands
+func NewRootCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "starq [<config-file>.yaml...] [-r <jq-rule> ...] [-a <jq-rule> ...]",
+		Short: "starq transforms a document from stdin using jq-style rules.",
+		Long:  `starq (pronounced "star-q") transforms a JSON or YAML document from stdin using jq-style rules and outputs to stdout.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := cmd.ParseFlags(args)
+			if err != nil {
+				return err
+			}
 
-		var opts starq.Opts
-		opts.ConfigFile, err = cmd.Flags().GetString(flagConfigFile)
-		exitIf(err)
-		opts.PrependRules, err = cmd.Flags().GetStringArray(flagPrependRule)
-		exitIf(err)
-		opts.AppendRules, err = cmd.Flags().GetStringArray(flagAppendRule)
-		exitIf(err)
+			var opts starq.Opts
+			opts.ConfigFiles = cmd.Flags().Args()
+			opts.PrependRules, err = cmd.Flags().GetStringArray(flagPrependRule)
+			if err != nil {
+				return err
+			}
+			opts.AppendRules, err = cmd.Flags().GetStringArray(flagAppendRule)
+			if err != nil {
+				return err
+			}
```

Comment:
I think based on the codecov, its not tested. I think can you can test by perhaps sending a wrong data type like int or bool. Not sure, if we run it on terminal, if that will ever happen.